src/libvirtApi/domain.js

```diff
@@ -770,41 +707,40 @@ export function domainGetCapabilities({ connectionName, arch, model }) {
-         * - /var/log/libvirt/qemu/ - For privileged (qemu:///system) VMs
-         * - ~/.cache/libvirt/qemu/logs - For non-privileged  (qemu:///session) VMs
-         */
-        const logFile = connectionName === "system" ? `/var/log/libvirt/qemu/${vmName}.log` : `${loggedUser.home}/.cache/libvirt/qemu/log/${vmName}.log`;
+    const loggedUser = await cockpit.user();
+    const options = { err: "message" };
+    if (connectionName === "system")
+        options.superuser = "try";
 
+    /* The possible paths of logfiles path, as of libvirt 9.3.0 are defined in:
+        * https://gitlab.com/libvirt/libvirt/-/blob/v9.3.0/src/qemu/qemu_conf.c#L153
+        * There libvirt defines 3 possible directories where they can be located
+        * - /root/log/qemu/ - That is however only relevant for "qemu:///embed", and not cockpit's use case
+        * - /var/log/libvirt/qemu/ - For privileged (qemu:///system) VMs
+        * - ~/.cache/libvirt/qemu/logs - For non-privileged  (qemu:///session) VMs
+        */
+    const logFile = connectionName === "system" ? `/var/log/libvirt/qemu/${vmName}.log` : `${loggedUser.home}/.cache/libvirt/qemu/log/${vmName}.log`;
+
+    try {
         // Use libvirt APIs for getting VM start up time when it's implemented:
         // https://gitlab.com/libvirt/libvirt/-/issues/481
-        return cockpit.script(`grep ': starting up' '${logFile}' | tail -1`, options);
-    })
-            .then(line => {
-                // Line from a log with a start up time is expected to look like this:
-                // 2023-05-05 11:22:03.043+0000: starting up libvirt version: 8.6.0, package: 3.fc37 (Fedora Project, 2022-08-09-13:54:03, ), qemu version: 7.0.0qemu-7.0.0-9.fc37, kernel: 6.0.6-300.fc37.x86_64, hostname: fedora
-
-                // Alternatively regex line.match(/\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}/g) can be used
-                const timeStr = line.split(': starting up')[0];
-                const date = new Date(timeStr);
-                return isNaN(date) ? null : date;
-            })
-            .catch(ex => {
-                console.log("Unable to detect domain start time:", ex);
-                return null;
-            });
+        const line = await cockpit.script(`grep ': starting up' '${logFile}' | tail -1`, options);
+
+        // Line from a log with a start up time is expected to look like this:
+        // 2023-05-05 11:22:03.043+0000: starting up libvirt version: 8.6.0, package: 3.fc37 (Fedora Project, 2022-08-09-13:54:03, ), qemu version: 7.0.0qemu-7.0.0-9.fc37, kernel: 6.0.6-300.fc37.x86_64, hostname: fedora
+
+        // Alternatively regex line.match(/\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}/g) can be used
+        const timeStr = line.split(': starting up')[0];
+        const date = new Date(timeStr);
+        return isNaN(date) ? null : date;
+    } catch (ex) {
+        console.log("Unable to detect domain start time:", ex);
+        return null;
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1397-20240122-152937-9774631b-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)